integration/helper/namespace.go

```diff
@@ -10,26 +14,76 @@ import (
 
 func TempNamespace(t *testing.T, client client.Client) *corev1.Namespace {
 	t.Helper()
+	// namespace ends up as "acorn-test-{random chars}"
+	return namedTempCreate(t, client, "acorn-test-", false, true)
+}
+
+func NamedTempProject(t *testing.T, client client.Client, name string) *corev1.Namespace {
+	t.Helper()
+	return namedTempCreate(t, client, name, true, false)
+}
 
+func namedTempCreate(t *testing.T, client client.Client, name string, project bool, generate bool) *corev1.Namespace {
+	t.Helper()
+	namespaceLabels := map[string]string{
+		"test.acorn.io/namespace": "true",
+	}
+	if project {
+		namespaceLabels[labels.AcornProject] = "true"
+	}
+	// double check to match actual project creation spec.
+	// Might be fitting to the test
 	ns := corev1.Namespace{
 		ObjectMeta: metav1.ObjectMeta{
-			GenerateName: "acorn-test-",
-			Labels: map[string]string{
-				"test.acorn.io/namespace": "true",
-			},
+			Labels: namespaceLabels,
 		},
 		Spec:   corev1.NamespaceSpec{},
 		Status: corev1.NamespaceStatus{},
 	}
+
+	if generate {
+		ns.ObjectMeta.GenerateName = name
+	} else {
+		ns.ObjectMeta.Name = name
+		ns.ObjectMeta.Namespace = name
+	}
+
```

Comment:
What if you just took in `metav1.ObjectMeta` here and applied it to the namespace when you create it? That would remove the name, project and generate parameters as well as some of this logic. It also has the benefit of being flexible should we need to add different labels or annotations.

Reply:
Great point, that cleaned it up a bit. Thank you!